### PR TITLE
BZ #1931 - New "Atomic Measurement" Resource Type

### PR DESCRIPTION
**Pull request by mjkoster**
Added  the file oic.wk.col.atomic.raml

### CODE REVIEW

File: oic.wk.col.atomic.raml [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+      Measurement must be hosted locally, that is on the same platform as the
+      Atomic Measurement resource itself.
+
+  - title: Atomic Measurement Restrictions on included resources
+    content: |
+      Atomic Measurement restricts properties to be exposed only through the
+      oic.if.b interface of the Atomic Measurement, thus insuring that all
+      of the items are accessed "atomically" as a batch.
+
+      If a client attempts to access an item in an Atomic Measurement individually,
```
Comment (rabardini) [inline oic.wk.col.atomic.raml]: Remove descriptive text that is a duplicate of the Core spec.  Also do not put any "normative" text in comments.

File: oic.wk.col.atomic.raml [hunk 2]
```diff
@@ -0,0 +1,238 @@
...
+      for each interface which is supported by the collection.
+
+      These three URIs are defined as the same URI path, only distinguished
+      by the different query parameter traits.
+
+schemas:
+  - baseline: !include schemas/oic.collection-schema.json
+  - batch-retrieve: !include schemas/oic.collection.batch-retrieve-schema.json
+  - links: !include schemas/oic.collection.linkslist-schema.json
+  - diag-payload: !include schemas/oic.diag-payload-schema.json
```
Comment (rabardini) [inline oic.wk.col.atomic.raml]: Remove oic.diag-payload-schema.json

File: oic.wk.col.atomic.raml [hunk 3]
```diff
@@ -0,0 +1,238 @@
...
+                  {
+                    "href": "timestamp",
+                    "rt":   ["oic.r.time.observed"],
+                    "if":   ["oic.if.s", "oic.if.baseline"],
+                  }
+                ]
+              }
+
+
+  post:
```
Comment (rabardini) [inline oic.wk.col.atomic.raml]: Remove "post" from RAML file as UPDATE is prohibited.

File: oic.wk.col.atomic.raml [hunk 4]
```diff
@@ -0,0 +1,238 @@
...
+    displayName: Batch Representation
+    is: [ interface-b ]
+
+    get:
+      description: |
+        Retrieve on Batch Interface
+
+      responses:
+        200:
+          description: "All targets returned OK status (HTTP 200 or CoAP 2.05 Content)"
```
Comment (rabardini) [inline oic.wk.col.atomic.raml]: Reword description and do not put quotes in the description text.

File: oic.wk.col.atomic.raml [hunk 5]
```diff
@@ -0,0 +1,238 @@
...
+                    "rep":
+                      {
+                        "bloc": "ear"
+                      }
+                  },
+                  {
+                    "href": "timestamp",
+                    "rep":
+                      {
+                        "datetime": "2007-04-05T14:30+09:00"
```
Comment (rabardini) [inline oic.wk.col.atomic.raml]: datetime -> observedtime

File: oic.wk.col.atomic.raml [hunk 6]
```diff
@@ -0,0 +1,238 @@
...
+                      }
+                  }
+                ]
+        500:
```
Comment (rabardini) [inline oic.wk.col.atomic.raml]: Remove the 500 response block.

File: oic.wk.col.atomic.raml [hunk 7]
```diff
@@ -0,0 +1,238 @@
...
+              schema: diag-payload
+              example: |
+                "Internal Server Error"
+
+
+/AtomicMeasurementURI?if=oic.if.ll:
+    description: |
+      The oic.if.ll interface exposes a representation of the links
+
+    displayName: Batch Representation
```
Comment (rabardini) [inline oic.wk.col.atomic.raml]: should be linked list representation not batch representation

**Comment by mjkoster** [comment]
Removed all of the normative and quasi-normative text as per the above comment

File: oic.wk.atomicmeasurement.raml [hunk 8]
```diff
@@ -0,0 +1,178 @@
...
+    content: !include LICENSE.md
+  - title: OCF Disclaimer
+    content: !include DISCLAIMER.md
+  - title: Atomic Measurement
+    content: |
+      Atomic Measurement is a specialization of the collection pattern to insure
+      that the linked resources are only accessed together.
+
+      Atomic Measurement exposes properties only through the oic.if.b interface,
+      thus ensuring that all of the items are accessed "atomically" in the batch.
```
Comment (mkettrick) [inline oic.wk.atomicmeasurement.raml]: Actually, you can find the Properties of the AM Resource itself using baseline or ll, so this needs to be reworded slightly to state the it's talking about the Properties of the linked Resources.

File: oic.wk.atomicmeasurement.raml [hunk 9]
```diff
@@ -0,0 +1,178 @@
...
+                    "rep":
+                      {
+                        "bloc": "ear"
+                      }
+                  },
+                  {
+                    "href": "timestamp",
+                    "rep":
+                      {
+                        "observedtime": "2007-04-05T14:30+09:00"
```
Comment (mkettrick) [inline oic.wk.atomicmeasurement.raml]: Did the Property Name of the timestamp Resource change to something else other than "observedtime"?  I can't remember but I recall talking about it when reviewing the HC data models.

File: oic.wk.atomicmeasurement.raml [hunk 10]
```diff
@@ -0,0 +1,178 @@
...
+      Atomic Measurement is a specialization of the collection pattern to insure
+      that the linked resources are only accessed together.
+
+      Atomic Measurement exposes properties only through the oic.if.b interface,
+      thus ensuring that all of the items are accessed "atomically" in the batch.
+
+  - title: Resource Design
+    content: |
+      The Atomic Measurement resource is specified as three Base URIs, one
+      for each interface which is supported by the collection.
```
Comment (mkettrick) [inline oic.wk.atomicmeasurement.raml]: "by the collection" should be change to "by the Atomic Measurement"

File: oic.wk.atomicmeasurement.raml [hunk 11]
```diff
@@ -0,0 +1,178 @@
...
+      200:
+        body:
+          application/json:
+            schema: baseline
+            example: |
+              {
+                "rt": ["oic.wk.col.atomic"],
+                "if": ["oic.if.b", "oic.if.ll", "oic.if.baseline"]
+                "rts": [ "oic.r.temperature", "oic.r.body.location.temperature", "oic.r.time.observed" ],
+                "links": [
```
Comment (mkettrick) [inline oic.wk.atomicmeasurement.raml]: Should we add "rts-m" here as well?

File: oic.wk.atomicmeasurement.raml [hunk 12]
```diff
@@ -0,0 +1,178 @@
...
+      Retrieve on Baseline Interface
+
+    responses:
+      200:
+        body:
+          application/json:
+            schema: baseline
+            example: |
+              {
+                "rt": ["oic.wk.col.atomic"],
```
Comment (mkettrick) [inline oic.wk.atomicmeasurement.raml]: This is the wrong "rt" value, right?  I think it was changed to oic.wk.atomicmeasurement.

File: oic.wk.atomicmeasurement.raml [hunk 13]
```diff
@@ -0,0 +1,178 @@
...
+                    "if":   ["oic.if.s", "oic.if.baseline"],
+                  },
+                  {
+                    "href": "bodylocation",
+                    "rt":   ["oic.r.body.location.temperature"],
+                    "if":   ["oic.if.s", "oic.if.baseline"],
+                  },
+                  {
+                    "href": "timestamp",
+                    "rt":   ["oic.r.time.observed"],
```
Comment (mkettrick) [inline oic.wk.atomicmeasurement.raml]: I can't find a schema for this but I did find one for oic.r.time.stamp.  Did that new schema replace this one?

File: oic.wk.atomicmeasurement.raml (hunk 9 above)
Comment (mkettrick) [reply oic.wk.atomicmeasurement.raml]: If this is in fact oic.r.time.stamp, then the Property name is now called "timestamp".

**Review by jacekhryszkiewicz — CHANGES_REQUESTED** [review]
Missing comma in line 77

**Pull request merged**